nncf/quantization/algorithms/channel_alignment/backend.py

```diff
@@ -0,0 +1,150 @@
+    conv_weight_in_channels_dim: int
+    bias_channels_dim: int
+
+
+class ChannelAlignmentAlgoBackend:
+    @staticmethod
+    def target_point(target_type: TargetType, target_node_name: str, port_id: int) -> TargetPoint:
+        """
+        Returns backend-specific target point.
+
+        :param target_type: Type of the location that should be modified.
+        :param target_node_name: Name of the located node.
+        :param port_id: id of the port for the statistics disctribution.
+        :return: Backend-specific TargetPoint.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_bias_value(node: NNCFNode, model: TModel, nncf_graph: NNCFGraph) -> np.ndarray:
+        """
+        Returns bias value in the NumPy format of provided node.
+
+        :param node: Node of NNCFGraph with bias value.
+        :param model: Backend-specific model for the initializer finding.
+        :param nncf_graph: NNCFGraph instance with the node.
+        :return: Bias value in the NumPy format.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_weight_value(node: NNCFNode, model: TModel, port_id: int) -> np.ndarray:
+        """
+        Returns bias value in the NumPy format of provided node.
+
+        :param node: Node of NNCFGraph with bias value.
+        :param model: Backend-specific model for the initializer finding.
+        :param nncf_graph: NNCFGraph instance with the node.
+        :return: Bias value in the NumPy format.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_activation_port_ids_for_node(node: NNCFNode) -> Tuple[int, int]:
+        """
+        Returns Input Port ID and Output Port ID corresponding to activation input and output edges for
+        the node.
+        Supports only nodes that could have bias value.
+
+        :param node: Node of NNCFGraph with bias value.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_weights_port_ids_for_node(node: NNCFNode) -> Tuple[int, int]:
+        pass
+
+    @staticmethod
+    @abstractmethod
+    def get_statistic_collector(
+        reduction_shape, q: float, num_samples: int, inplace: bool
+    ) -> TensorStatisticCollectorBase:
+        pass
+
+    @staticmethod
+    @abstractmethod
+    def is_node_with_bias(node: NNCFNode, nncf_graph: NNCFGraph) -> bool:
+        pass
+
+    @staticmethod
+    @abstractmethod
+    def create_bias_update_command(node_with_bias: NNCFNode, updated_value: np.ndarray, nncf_graph: NNCFGraph):
+        pass
+
+    @staticmethod
+    @abstractmethod
+    def create_weights_update_command(node_with_weights: NNCFNode, updated_value: np.array, weights_port_id: int):
+        pass
+
+    @staticmethod
+    @abstractmethod
+    def get_dims_descriptor(node: NNCFNode) -> DimsDescriptor:
+        pass
+
+    @staticmethod
+    @abstractmethod
+    def get_conv_layer_attributes(node: NNCFNode) -> Optional[ConvolutionLayerAttributes]:
+        pass
+
+
+class ConvParamsContainer:
```

Comment:
Why is it in `backend.py`?

Reply:
I'll move it to algorithm.py

Reply:
I've moved it to `algorithm.py`, please check